js-packages/@prestojs/util/src/identifiable.ts

```diff
@@ -2,50 +2,82 @@ type SingleId = string | number;
 type CompoundId = { [fieldName: string]: SingleId };
 export type Id = SingleId | CompoundId;
 
+/**
+ * Defines a property `_pk` which is taken as the `id`. [ViewModel](doc:ViewModelFactory)
+ * always implements this interface.
+ */
+interface IdentifiablePk {
+    /**
+     * The id for this item
+     */
+    _pk: Id;
+}
+
+/**
+ * Defines a property `id` to uniquely identify the item
+ */
+interface IdentifiableId {
```

Comment:
I think it was really because `id` is the intuitive thing if implementing `Identifiable` it's just that it's not suitable for our use case (although would work most of the time). That's where I started from anyway but didn't want to force `ViewModel` to conform to this so added `_pk` as the support. 

I guess we could just change it to `_pk` it just sounds a bit more specific to a database implementation

(this isn't new behaviour in this PR BTW - I just refactored how the type was defined to make it easier to work with internally)